framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -539,6 +541,35 @@ protected void postDirectSuperTypes(AnnotatedTypeMirror type,
     }
 
     /**
+     * Gets the type of the resulting constructor call of a MemberReferenceTree.
+     *
+     * @param memberReferenceTree MemberReferenceTree where the member is a constructor
+     * @param constructorType     AnnotatedExecutableType of the declaration of the constructor
+     * @return AnnotatedTypeMirror of the resulting type of the constructor
+     */
+    public AnnotatedTypeMirror getResultingTypeOfConstructorMemberReference(MemberReferenceTree memberReferenceTree,
+                                                                            AnnotatedExecutableType constructorType) {
+        if (memberReferenceTree.getMode() != MemberReferenceTree.ReferenceMode.NEW) {
+            ErrorReporter.errorAbort("");
```

Comment:
Some useful message in case this ever gets hit. If this is unlikely, should it be an assert instead?